source/Xunit.Gherkin.Quick/CoreModel/FeatureFile/FeatureFile.cs

```diff
@@ -1,30 +1,40 @@
-﻿using Gherkin.Ast;
+﻿using System;
 using System.Linq;
+using System.Collections.Generic;
+using Gherkin.Ast;
+using System.Linq;
+
+using Scenario = global::Gherkin.Ast.Scenario;
+using Background = global::Gherkin.Ast.Background;
+using StepsContainer = Gherkin.Ast.StepsContainer;
 
 namespace Xunit.Gherkin.Quick
 {
     internal sealed class FeatureFile
     {
+        
         public GherkinDocument GherkinDocument { get; }
 
         public FeatureFile(GherkinDocument gherkinDocument)
         {
             GherkinDocument = gherkinDocument ?? throw new System.ArgumentNullException(nameof(gherkinDocument));
+
+            GherkinDialect.Register(
+                GherkinDocument.Feature?.Language,
+                GherkinDocument.Feature?.Location
+            );
```

Comment:
Instance constructor should not have logic like this. Move this code somewhere else. e.g., inside FeatureFileRepository where the FeatureFIle instance is created.